parsec/backend/asgi/rpc.py

```diff
@@ -205,8 +206,16 @@ async def _do_handshake(
         except CryptoError:
             _handshake_abort(401, api_version=api_version)
 
-    if raw_invitation_token:
-        token = InvitationToken.from_hex(raw_invitation_token)
+    if not check_invitation:
+        invitation = None
+    else:
+        raw_invitation_token = request.headers.get("Invitation-Token", "")
+
+        try:
+            token = InvitationToken.from_hex(raw_invitation_token)
```

Comment:
You won't get away like this, you know we need to add a test for this !

Reply:
Done